src/Microsoft.AspNetCore.Mvc.TagHelpers/PartialTagHelper.cs

```diff
@@ -63,18 +73,45 @@ public class PartialTagHelper : TagHelper
                 throw new ArgumentNullException(nameof(context));
             }
 
+            var model = ResolveModel();
             var viewBuffer = new ViewBuffer(_viewBufferScope, Name, ViewBuffer.PartialViewPageSize);
             using (var writer = new ViewBufferTextWriter(viewBuffer, Encoding.UTF8))
             {
-                await RenderPartialViewAsync(writer);
+                await RenderPartialViewAsync(writer, model);
 
                 // Reset the TagName. We don't want `partial` to render.
                 output.TagName = null;
                 output.Content.SetHtmlContent(viewBuffer);
             }
         }
 
-        private async Task RenderPartialViewAsync(TextWriter writer)
+        // Internal for testing
+        internal object ResolveModel()
+        {
+            if (Model != null & For != null)
```

Comment:
Could you add a test for this? Also, maybe the error message might not "the"

> Cannot use ~~the~~ '{0}' with both '{1}' and '{2}' attributes

Reply:
Already have a test but can definitely update the error message.